docs/tutorial/agent-on-docker.md

```diff
@@ -0,0 +1,157 @@
+
+- docker: v19.0 ~
+- go: v1.14 ~
+- libhdf5 (_only required for this tutorial._)
+
+
+Hdf5 is required for this tutorial. If hdf5 is not installed, please install [hdf5](https://www.hdfgroup.org/).
+<details><summary>[Optional] Install hdf5</summary><br>
+
+```bash
+# yum
+yum install -y hdf5-devel
+
+# apt
+apt-get install libhdf5-serial-dev
+
+# homebrew
+brew install hdf5
+```
+</details>
+
+## Deploy
+
+This chapter will show you how to deploy Vald Agent on docker.<br>
+This chapter will use NGT for the core engine of Vald Agent.
+
+1. Clone the vdaas/vald repository
+
+    ```bash
+    git clone https://github.com/vdaas/vald.git
+    ```
+
+1. Create `config.yaml`
+
+    The configuration of Vald agent for docker is set using `config.yaml`<br>
+    You can also check [the sample configuration](https://github.com/vdaas/vald/blob/master/cmd/agent/core/ngt/sample.yaml).
+
+    ```bash
+    cat << EOF > config.yaml
+    ---
+    version: v0.0.0
+    time_zone: JST
+    logging:
+      logger: glg
+      level: debug
+      format: json
+    server_config:
+      servers:
+        - name: agent-grpc
+          host: 0.0.0.0
+          port: 8081
+          mode: GRPC
+          probe_wait_time: "3s"
+          http:
+            shutdown_duration: "5s"
+            handler_timeout: ""
+            idle_timeout: ""
+            read_header_timeout: ""
+            read_timeout: ""
+            write_timeout: ""
+      startup_strategy:
+        - agent-grpc
+      shutdown_strategy:
+        - agent-grpc
+      full_shutdown_duration: 600s
+      tls:
+        enabled: false
+        # cert: /path/to/cert
+        # key: /path/to/key
+        # ca: /path/to/ca
+    ngt:
+      # vector dimension
+      dimension: 784
+      # bulk insert chunk size
+      bulk_insert_chunk_size: 10
+      # distance_type, which should be "l1", "l2" "angle", "hamming", "cosine", "normalizedangle" or "nomralizedcosine"
+      distance_type: l2
+      # object_type, which should be "float" or "uint8"
+      object_type: float
+      # creation edge size
+      creation_edge_size: 20
+      # search edge size
+      search_edge_size: 10
+      # in-memory mode enabled
+      enable_in_memory_mode: true
+      # The limit duration of automatic indexing 
+      auto_index_duration_limit: 1m
+      # Check duration of automatic indexing.
+      auto_index_check_duration: 10s
+      # The number of cache to trigger automatic indexing
+      auto_index_length: 100
+    EOF
+    ```
+
+1. Deploy Vald Agent on Docker
+
+    To deploy Vald agent on docker with `config.yaml`, you can run below command.
+
+    ```bash
+    docker run -v $(pwd)/config.yaml:/etc/server/config.yaml -p 8081:8081 --rm -it vdaas/vald-agent-ngt
+    ```
+
+1. Verify
+
+    If the deployment success, you can confirm the output will be similar to below.
+
+    ```bash
+    {"Date":"2020-06-24T05:49:53.615263Z","Level":"INFO","Detail":"maxprocs: Leaving GOMAXPROCS=4: CPU quota undefined"}
```

Comment:
![badge](https://img.shields.io/badge/review-imo-orange.svg?style=flat-square)
I think 'raw' format of logging is more understandable.